// @storyblok/react/dist/index.mjs
import { useState as f, useEffect as k } from "react";
import { registerStoryblokBridge as u } from "./storyblok-js.mjs";
import { RichTextResolver as v, RichTextSchema as x, apiPlugin as R, loadStoryblokBridge as C, renderRichText as w, storyblokEditable as B } from "./storyblok-js.mjs";
import { useStoryblokApi as c } from "./index2.mjs";
import { getComponent as h, getCustomFallbackComponent as A, getEnableFallbackComponent as T, setComponents as F, storyblokInit as I } from "./index2.mjs";
import { useStoryblokState as P } from "./client.mjs";
import { default as J } from "./storyblok-component.mjs";
const b = (r, e = {}, o = {}) => {
  const s = c();
  if (!s)
    return console.error(
      "You can't use useStoryblok if you're not loading apiPlugin."
    ), null;
  let [l, n] = f({});
  o.resolveRelations = o.resolveRelations ?? e.resolve_relations, o.resolveLinks = o.resolveLinks ?? e.resolve_links;
  const i = typeof window < "u" && typeof window.storyblokRegisterEvent < "u";
  return k(() => {
    async function a() {
      const { data: t } = await s.get(
        `cdn/stories/${r}`,
        e
      );
      n(t.story), i && t.story.id && u(
        t.story.id,
        (y) => n(y),
        o
      );
    }
    a();
  }, [r, JSON.stringify(e)]), l;
};
export {
  v as RichTextResolver,
  x as RichTextSchema,
  J as StoryblokComponent,
  R as apiPlugin,
  h as getComponent,
  A as getCustomFallbackComponent,
  T as getEnableFallbackComponent,
  c as getStoryblokApi,
  C as loadStoryblokBridge,
  u as registerStoryblokBridge,
  w as renderRichText,
  F as setComponents,
  B as storyblokEditable,
  I as storyblokInit,
  b as useStoryblok,
  c as useStoryblokApi,
  u as useStoryblokBridge,
  P as useStoryblokState
};
